lib/handler/cache-handler.js

```diff
@@ -126,15 +126,25 @@ class CacheHandler extends DecoratorHandler {
         cacheControlDirectives
       )
 
-      this.#writeStream = this.#store.createWriteStream(this.#requestOptions, {
+      /**
+       * @type {import('../../types/cache-interceptor.d.ts').default.CacheStoreValue}
+       */
+      const value = {
         statusCode,
         statusMessage,
         rawHeaders: strippedHeaders,
         vary: varyDirectives,
         cachedAt: now,
         staleAt,
         deleteAt
-      })
+      }
+
+      // Check for the response's etag header, ignore if there's multiple
+      if (typeof headers.etag === 'string') {
+        value.etag = headers.etag
```

Comment:
Shouldn't etag be validated here? https://httpwg.org/specs/rfc9110.html#field.etag

Reply:
and also account for weak etags